First attempt to have CI-controlled process of releasing PROD image (#24433)
mistercrunch/Airflow: dev/README_RELEASE_AIRFLOW.md

Production Docker images should be manually prepared and pushed by the release manager or another committer
who has access to Airflow's DockerHub. Note that we started releasing a multi-platform build, so you need
to have an environment prepared to build multi-platform images. You can achieve it with:

* GitHub Actions Manual Job (easiest)
* Emulation (very slow)
* Hardware builders if you have both AMD64 and ARM64 hardware locally

> Building the image is triggered By running the `Release PROD image` workflow via

Reviewer: ```suggestion
Building the image is triggered by running the `Release PROD image` workflow via
```

> [GitHub Actions](https://github.com/apache/airflow/actions)

Reviewer: ```suggestion
[GitHub Actions](https://github.com/apache/airflow/actions).
```

When you trigger it you need to pass:

* Airflow Version

> * Additional "true" in skip field if you do not release latest image

Reviewer: ```suggestion
* Optional "true" in skip latest field if you do not want to retag the latest image
```

Reviewer: Since we are talking RC images here, should we change this to advise setting this to true?

## Manually prepare production Docker Image

> Building the image is triggered By running the `Release PROD image` workflow via
> [GitHub Actions](https://github.com/apache/airflow/actions)

Reviewer: ```suggestion
Building the image is triggered by running the `Release PROD image` workflow via
[GitHub Actions](https://github.com/apache/airflow/actions).
```

When you trigger it you need to pass:

* Airflow Version

> * Additional "true" in skip field if you do not release latest image

Reviewer: ```suggestion
* Optional "true" in skip latest field if you do not want to retag the latest image
```
